google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDisplayUploadAd.java

```diff
@@ -0,0 +1,222 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.ads.googleads.examples.advancedoperations;
+
+import com.beust.jcommander.Parameter;
+import com.google.ads.googleads.examples.utils.ArgumentNames;
+import com.google.ads.googleads.examples.utils.CodeSampleParams;
+import com.google.ads.googleads.lib.GoogleAdsClient;
+import com.google.ads.googleads.v3.common.AdMediaBundleAsset;
+import com.google.ads.googleads.v3.common.DisplayUploadAdInfo;
+import com.google.ads.googleads.v3.common.MediaBundleAsset;
+import com.google.ads.googleads.v3.enums.AdGroupAdStatusEnum.AdGroupAdStatus;
+import com.google.ads.googleads.v3.enums.AssetTypeEnum.AssetType;
+import com.google.ads.googleads.v3.enums.DisplayUploadProductTypeEnum.DisplayUploadProductType;
+import com.google.ads.googleads.v3.errors.GoogleAdsError;
+import com.google.ads.googleads.v3.errors.GoogleAdsException;
+import com.google.ads.googleads.v3.resources.Ad;
+import com.google.ads.googleads.v3.resources.AdGroupAd;
+import com.google.ads.googleads.v3.resources.Asset;
+import com.google.ads.googleads.v3.services.AdGroupAdOperation;
+import com.google.ads.googleads.v3.services.AdGroupAdServiceClient;
+import com.google.ads.googleads.v3.services.AssetOperation;
+import com.google.ads.googleads.v3.services.AssetServiceClient;
+import com.google.ads.googleads.v3.services.MutateAdGroupAdsResponse;
+import com.google.ads.googleads.v3.services.MutateAssetsResponse;
+import com.google.ads.googleads.v3.utils.ResourceNames;
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.ByteStreams;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.BytesValue;
+import com.google.protobuf.StringValue;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.net.URL;
+
+/** Adds a display upload ad to a given ad group. To get ad groups, run GetAdGroups.java. */
+public class AddDisplayUploadAd {
+
+  private static final String BUNDLE_URL = "https://goo.gl/9Y7qI2";
+
+  private static class AddDisplayUploadAdParams extends CodeSampleParams {
+
+    @Parameter(names = ArgumentNames.CUSTOMER_ID, required = true)
+    private Long customerId;
+
+    @Parameter(names = ArgumentNames.AD_GROUP_ID, required = true)
+    private Long adGroupId;
+  }
+
+  public static void main(String[] args) throws IOException {
+    AddDisplayUploadAdParams params = new AddDisplayUploadAdParams();
+    if (!params.parseArguments(args)) {
+
```

Comment:
Nit: extra blank line between brace and comment; doesn't seem consistent with rest of code, e.g. the catch below on line 87.

Reply:
The extra line in this specific line is consistent with all of the other Java code examples.